src/cmd/event.js

```diff
@@ -0,0 +1,196 @@
+    if (beautify) {
+        console.table(events);
+    } else {
+        console.log(events);
+    }
+}
+
+async function check_for_event_file() {
+    const path = '/var/log/noobaa_events.log';
+    return await fs_utils.file_exists(path);
+}
+
+async function get_active_events() {
+    const events_info = await get_all_events();
+    let active_events = [];
+    events_info.forEach(function(obj, index) {
+        if (obj) {
+            const event = JSON.parse(obj.toString());
+            const message = JSON.parse(event.message.toString());
+            message.message_thread = [];
+            const event_time = new Date(event.timestamp);
+            const time_diff = Date.now() - event_time.getTime();
+            if (time_diff < config.EVENT_ACTIVE_INTERVAL) {
+                message.time = new Date(event_time.getTime());
+                const repeated = structure_repeated_events(message, active_events);
+                if (!repeated || active_events.length === 0) {
+                    message.message_thread.push(message.message);
+                    active_events.push(message);
+                }
+            } else {
+                active_events = sort_events(active_events);
+                return active_events;
+            }
+        }
+    });
+    active_events = sort_events(active_events);
+    return active_events;
+}
+
+function structure_repeated_events(message, active_events) {
+    let repeated = false;
+    active_events.forEach(function(obj, index) {
+        if (obj.code === message.code) {
+            obj.occurrence = obj.occurrence ? obj.occurrence + 1 : 2;
+            if (obj.message_thread) {
+                obj.message_thread.push(message.message);
+            } else {
+                obj.message_thread = [message.message];
+            }
+            repeated = true;
+        }
+    });
+    return repeated;
+}
+
+async function list_events(time, severity) {
+    const events_info = await get_all_events();
+    const query_time = new Date(time);
+    let result_events = [];
+    events_info.forEach(function(obj, index) {
+        if (obj) {
+            const event = JSON.parse(obj.toString());
+            const message = JSON.parse(event.message.toString());
+            const event_time = new Date(event.timestamp);
+            message.time = event_time;
+            result_events.push(message);
+            if (event_time < query_time) {
+                result_events.pop();
+            }
+            if (severity && message.severity !== severity) {
+                result_events.pop();
+            }
+        }
+    });
+    result_events = await sort_events(result_events);
+    return result_events;
+}
+
+async function get_all_events() {
+    try {
+        const events_logs = await fs.promises.readFile('/var/log/noobaa_events.log');
+        const content = events_logs.toString();
+        let events_info = content.split('\n');
+        events_info = events_info.slice().reverse();
```

Comment:
why reversing the events?

Reply:
@romayalon Since we are reading from a log file,  most recent(active) event will be placed end of the file and to get the active events fast I thought reversing the array is a good option, another option is to traverse from the bottom. WDYT?

Reply:
it's obviously nice but I'm not sure if it's too much effort,
I checked the events file we have on a regular must gather and I don't see that it's sorted/reversed
@dannyzaken WDYT?

Reply:
Yeah, if we can avoid the reversal then we can also increase the log file size and just read the file line by line instead of reading it all in memory.